accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/AccessioningService.java

```diff
@@ -27,18 +27,39 @@
 
 import java.util.List;
 
+/**
+ * Service for retrieval and modifications of object accessions.
+ *
+ * @param <MODEL> Type of the objects identified by the accessions
+ * @param <HASH> Type of the hash calculated based on the fields that uniquely identify an accessioned object
+ * @param <ACCESSION> Accession of the object that identify a particular model
+ */
 public interface AccessioningService<MODEL, HASH, ACCESSION> {
 
+    /**
+     * Find the accessions associated with a list of objects.
+     * Searches object's accession in the repository, and if it does not exist, new accession is generated and stored in repository
+     *
+     * @param messages List of messages to be accessioned
```

Comment:
`List of objects, either accessioned or to be accessioned`? `messages` just happens to be the name of the argument.

Reply:
Agree, both are possible as this is getOrCreate `List of messages to be accessioned or already accessioned`